lib/Sema/TypeCheckNameLookup.cpp

```diff
@@ -244,6 +244,106 @@ TypeChecker::lookupUnqualifiedType(DeclContext *dc, DeclName name,
       decls.push_back(cast<TypeDecl>(found.getValueDecl()));
   }
 
+  // If we're looking for a type named `CodingKeys`, then we've got some
+  // additional work to do here. CodingKeys is a special type: although not a
+  // protocol requirement of the Encodable/Decodable protocols, it is
+  // synthesized as part of Encodable/Decodable conformance. Unqualified lookup
+  // may occur before the type is synthesized, though, in which case, we need to
+  // make sure it gets synthesized.
+  //
+  // The unqualified lookups above, however, cannot force Encodable/Decodable
+  // synthesis to occur (since they know nothing about typechecking). Thus, we
+  // also need to find the closest qualified type which might contain a
+  // CodingKeys type and synthesize it.
```

Comment:
Sure, I can do that. That felt like a more sweeping change for this point in the release, but I'll try to refactor into a form like that.